packages/wdio-browserstack-service/src/testOps/requestUtils.ts

```diff
@@ -0,0 +1,47 @@
+import type { UploadType } from '../types.js'
+import { DATA_ENDPOINT, DATA_EVENT_ENDPOINT, DATA_SCREENSHOT_ENDPOINT } from '../constants.js'
+import { BStackLogger } from '../bstackLogger.js'
+import { DEFAULT_REQUEST_CONFIG, getLogTag } from '../util.js'
+import got from 'got'
+
+export async function uploadEventData (eventData: UploadType | Array<UploadType>, eventUrl: string = DATA_EVENT_ENDPOINT) {
+    let logTag: string = 'BATCH_UPLOAD'
+    if (!Array.isArray(eventData)) {
+        logTag = getLogTag(eventData.event_type)
+    }
+
+    if (eventUrl === DATA_SCREENSHOT_ENDPOINT) {
+        logTag = 'screenshot_upload'
+    }
+
+    if (!process.env.BS_TESTOPS_BUILD_COMPLETED) {
+        return
+    }
+
+    if (!process.env.BS_TESTOPS_JWT) {
+        BStackLogger.debug(`[${logTag}] Missing Authentication Token/ Build ID`)
+        return {
+            status: 'error',
+            message: 'Token/buildID is undefined, build creation might have failed'
+        }
+    }
+
+    try {
+        const url = `${DATA_ENDPOINT}/${eventUrl}`
+        const data = await got.post(url, {
+            agent: DEFAULT_REQUEST_CONFIG.agent,
+            headers: {
+                ...DEFAULT_REQUEST_CONFIG.headers,
+                'Authorization': `Bearer ${process.env.BS_TESTOPS_JWT}`
+            },
+            json: eventData
+        }).json()
+        BStackLogger.debug(`[${logTag}] Success response: ${JSON.stringify(data)}`)
+    } catch (error) {
+        BStackLogger.debug(`[${logTag}] Failed. Error: ${error}`)
```

Comment:
Shouldn't we return an error object here too similar to line 23?

Reply:
Actually this method should throw error for any reason it is not able to make the request, forgot to update it before. Have updated it now to throw errors